### ISSUE TITLE
[Title] Should NixOS/nixpkgs adjust its platform support?

### ISSUE THREAD

**Issue by lunik1** [NixOS/SC-election-2024#106]
### Question

Of the currently supported platforms, are there any you think should be promoted to a higher tier or demoted to a lower tier? Are there any currently entirely unsupported platforms which should gain some level of support?

(bit of overlap with #97 with re: darwin specifically)

### Reminder of the Q&A rules

Please adhere to the [Q&A guidelines and rules](https://github.com/NixOS/SC-election-2024/blob/main/doc/qna.md)

**Comment by getchoo** [comment]
> Of the currently supported platforms, are there any you think should be promoted to a higher tier or demoted to a lower tier?

No, I think we're in a good spot currently. I would *like* to see more platforms promoted to tier 1, but I believe that requires improving their existing support (read: Darwin support) before really considering that

> Are there any currently entirely unsupported platforms which should gain some level of support?

I am a [little bit bias here](https://www.github.com/NixOS/nixpkgs/pull/335067), but aarch64-windows is an important platform to grow support for as Windows on Arm machines become more mainstream

**Comment by proofconstruction** [comment]
I've answered some of this in #97 in the context of darwin support, but I think there's also value in eventually targeting the Power ISA[1] (really!), and I'd like to see RISC-V[2] support happen sooner than later.

[1] I say this partly because it's a bit funny to think about, but also because a lot of important business logic runs on POWER, and probably will for the forseeable future. If we could bring the much higher degree of environmental control we normally enjoy to contexts like healthcare information systems and bank transaction engines, I'm sure we could deliver (and capture!) a huge amount of value across the global economy.

[2] RISC-V is the new hotness, and there's already a tremendous amount of energy behind its development. You can expect to see some very interesting things happen in the next few years. Look to the east.

**Comment by jtojnar** [comment]
I think we should actually [implement](https://nixos.org/manual/nixpkgs/unstable/#chap-platform-support) the [RFC46](https://github.com/NixOS/rfcs/pull/46) and then periodically re-evaluate if the platforms satisfy the criteria. According to that, I think Darwin should be demoted but it can easily be promoted again once the ofborg situation is improved.
